HULKs/hulk
# Fix parallel camera access die zweite

## Introduced Changes

Fixes the camera access bug. Also some minor refactoring in the camera, to remove one layer of interior mutability.

Fixes #961 
Fixes #736

## ToDo / Known Issues

*If this is a WIP describe which problems are to be fixed.*

## Ideas for Next Iterations (Not This PR)

*If there are some improvements that could be done in a next iteration, describe them here.*

## How to Test

- Upload to NAO (should never crash)
- check vision recordings, should never process same frame twice
